### ISSUE TITLE
Compilation fails when using musl

### ISSUE THREAD

**Issue by dsprenkels** [dsprenkels/randombytes#11]
@jaromil gets [compilation errors](https://travis-ci.org/DECODEproject/zenroom/jobs/358002125#L2621-L2632) when compiling with the musl libc:

```
randombytes.c: In function ‘randombytes_linux_get_entropy_avail’:
randombytes.c:134:12: error: ‘RNDGETENTCNT’ undeclared (first use in this function)
  ioctl(fd, RNDGETENTCNT, &ret);
            ^
randombytes.c:134:12: note: each undeclared identifier is reported only once for each function it appears in
```

Note that compilation also fails on `getrandom`, because `linux/random.h` cannot be found (header file redundant, because of `sys/syscall.h`?).

- Musl does not seem to be able to import linux symbols (because of portability issues). They say they will support `getrandom()` (not `SYS_getrandom`) here: http://www.openwall.com/lists/musl/2017/09/28/8. But in my case musl (`1.1.19-1`) hasn't yet added support.
- In these cases we should indeed fall back to using `/dev/urandom`. But `RNDGETENTCNT` is (probably) glib-specific. I hesitate to skip the entropy test. Maybe see what libsodium does here.

- [ ] `getrandom` syscall fix
- [ ] `/dev/urandom` fix

**Comment by dsprenkels** [comment]
Libsodium does not seem to actually check the entropy level of the pool. Instead, they block on `/dev/random` once and assume that after the kernel passes back control, the pool is sufficiently filled.

I am personally not sure about the entropy levels in this case (did not find any documentation), so to me it seems that we have two options:

1. Create a fallback to `RNDGETENTCNT`, which reads from `/proc/sys/kernel/random/entropy_avail`.
2. Do not use `RNDGETENTCNT` at all, but only read from `/proc/sys/kernel/random/entropy_avail`.
3. Block on `/dev/random` once (using a global value to make sure we don't block again).
4. Block on `/dev/random` on every call to `randombytes`.

I would personally opt for 1., with a fallback that if `/proc/sys/kernel/random/entropy_avail` cannot be read, we will block on `/dev/random` (every time).

**Comment by jaromil** [comment]
You may not like what you see :^) but this is how I solved it (for now?)
https://github.com/DECODEproject/zenroom/blob/develop/src/randombytes.c
this builds correctly on all zenroom targets (musl,osx,win,arm-gcc,arm-musl,emscripten)

**Comment by dsprenkels** [comment]
Yeah, bypassing the entropy check is not an option. :P

**Comment by dsprenkels** [comment]
My current plan is to go for option 2, because it can be done with any C library in a portable manner.

**Comment by dsprenkels** [comment]
5. Define `RNDGETENTCNT` inline.

**Comment by jaromil** [comment]
my favorites are 1. and 2. - thanks for taking care of details :^)

**Comment by dsprenkels** [comment]
Is fixed by #12. Used option 5.
